torch/fx/experimental/symbolic_shapes.py

```diff
@@ -122,6 +122,8 @@ def expr(self):
         return self._expr
 
     def wrap(self, num):
+        if not isinstance(num, int):
```

Comment:
This is a new invariant added here to catch bugs where we would get `SymIntNode(PySymFloat(..))`. Is that ok?

Reply:
This seems good to me - although if we're gonna add the asserts here, what do you think of additionally adding them to the `PythonSymIntNodeImpl` and `PythonSymFloatNodeImpl` constructors? e.g. that the `PythonSymIntNodeImpl` class should only ever hold a PyObject corresponding to an int64_t, or a sym int.

Reply:
Forcing SymFooNode to only contain PySymFoo sounds good.
For the content of the PySymFoo, I'm not sure (as discussed below) as we need to at least support foo and bool.
Leaving that for a follow up.